### PR TITLE
tracking controller and render events

### PR DESCRIPTION
**Pull request by epilgrim**
This change is [Reviewable](https://reviewable.io/reviews/kagux/ex_debug_toolbar/11)

### CODE REVIEW

File: lib/ex_debug_toolbar/collector/instrumentation_collector.ex [hunk 1]
```diff
@@ -8,4 +8,31 @@ defmodule ExDebugToolbar.Collector.InstrumentationCollector do
   def ex_debug_toolbar(:stop, _, _) do
     Toolbar.finish_event("request")
   end
+
+  def phoenix_controller_call(:start, _, %{conn: conn}) do
+    event_name = controller_event_name(conn)
+    Toolbar.start_event(event_name)
```
Comment (kagux) [inline lib/ex_debug_toolbar/collector/instrumentation_collector.ex]: these two lines could be nicely piped

File: test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs [hunk 2]
```diff
@@ -19,11 +22,50 @@ defmodule ExDebugToolbar.Collector.InstrumentationCollectorTest do
...
+      call_collector(&Collector.ex_debug_toolbar/3, %{})
       assert {:ok, request} = get_request(@request_id)
       assert request.data.timeline.events |> Enum.any?
       assert request.data.timeline.duration > 0
     end
   end
+
+  describe "phoenix_controller_call"  do
+    test "it starts a phoenix_controller event" do
+      Toolbar.start_request
```
Comment (kagux) [inline test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs]: maybe something like this to clean up a bit 
```
setup :start_request

defp start_request do
 Toolbar.start_request
end
```

`start_request/0` could even be put to request helpers

File: test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs [hunk 3]
```diff
@@ -19,11 +22,50 @@ defmodule ExDebugToolbar.Collector.InstrumentationCollectorTest do
...
+      Collector.phoenix_controller_call(:start, %{}, %{conn: %{}})
+      assert {:ok, request} = get_request(@request_id)
+      assert request.data.timeline.queue |> Enum.count > 0
```
Comment (kagux) [inline test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs]: do we really want to assert the queue? I think in both cases having that second test that does start and stop is good enough.

File: test/support/data/timeline_helpers.ex [hunk 4]
```diff
@@ -0,0 +1,10 @@
+defmodule ExDebugToolbar.Test.Support.Data.TimelineHelpers do
+  alias ExDebugToolbar.Data.Timeline
+
+  def find_event(%Timeline{} = timeline, event_name), do: timeline.events |> find_event(event_name)
+  def find_event([%{name: event_name} = event | _], event_name), do: event
+  def find_event([event | events], event_name) do
+    find_event(event.events, event_name) || find_event(events, event_name)
+  end
+  def find_event(_, _), do: false
```
Comment (kagux) [inline test/support/data/timeline_helpers.ex]: wouldn't nil be better?

Comment (epilgrim) [inline test/support/data/timeline_helpers.ex]: indeed

File: lib/ex_debug_toolbar/collector/instrumentation_collector.ex (hunk 1 above)
Comment (epilgrim) [inline lib/ex_debug_toolbar/collector/instrumentation_collector.ex]: no, it cannot, as the function needs to return the event name

File: test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs (hunk 2 above)
Comment (epilgrim) [inline test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs]: already done :)

File: test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs (hunk 3 above)
Comment (epilgrim) [inline test/lib/ex_debug_toolbar/collector/instrumentation_collector_test.exs]: this one, i'm not sure. It depends on how much we trust ourselves :P 
are we good just asserting that if both of the methods are called in order, the event will be there in the timeline, without having to assert that one opens it, and the other closes it?

**Pull request merged**